src/mlpack/tests/nesterov_momentum_sgd_test.cpp

```diff
@@ -0,0 +1,81 @@
+/**
+ * @file nesterov_momentum_sgd_test.cpp
+ * @author Sourabh Varshney
+ *
+ * Test file for NesterovMomentumSGD (Stochastic gradient descent with
+ * nesterov momentum updates).
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#include <mlpack/core.hpp>
+#include <mlpack/core/optimizers/sgd/sgd.hpp>
+#include <mlpack/core/optimizers/sgd/update_policies/gradient_clipping.hpp>
+#include <mlpack/core/optimizers/sgd/update_policies/nesterov_momentum_update.hpp>
+#include <mlpack/core/optimizers/problems/generalized_rosenbrock_function.hpp>
+#include <mlpack/core/optimizers/problems/sgd_test_function.hpp>
+
+#include <boost/test/unit_test.hpp>
+#include "test_tools.hpp"
+
+using namespace std;
+using namespace arma;
+using namespace mlpack;
+using namespace mlpack::optimization;
+using namespace mlpack::optimization::test;
+
+BOOST_AUTO_TEST_SUITE(NesterovMomentumSGDTest);
+
```

Comment:
Can you add for each test a simple comment that describes the test, something like: https://github.com/mlpack/mlpack/blob/master/src/mlpack/tests/adam_test.cpp

Reply:
Added the comments. :)